src/map/mob.cpp

```diff
@@ -804,10 +804,10 @@ int mob_attack(dumb_ptr<mob_data> md, tick_t tick)
 
     nullpo_retz(md);
 
-    if ((tbl = map_id2bl(md->target_id)) == nullptr)
+    if (!mob_check_attack(md))
         return 0;
 
-    if (!mob_check_attack(md))
+    if ((tbl = map_id2bl(md->target_id)) == nullptr)
```

Comment:
No. mob_check_attack resets the target_id when ((tbl = map_id2bl(md->target_id)) == nullptr) fails.